drivers/ipm/ipm_cavs_idc.c

```diff
@@ -229,13 +229,15 @@ DEVICE_DT_INST_DEFINE(0, &cavs_idc_init, device_pm_control_nop,
 		    &cavs_idc_driver_api);
 
 #ifdef CONFIG_SCHED_IPI_SUPPORTED
-static int cavs_idc_smp_init(const struct device *dev)
+int cavs_idc_smp_init(const struct device *dev)
```

Comment:
I have a half-finished patch that moves the IDC handling into the main SOC code and out of a driver (leaving IPM implemented in terms of that SOC API).    That seems like a cleaner path for this sort of problem.  No reason not to merge this, but IMHO there's a better way long term.